### PR TITLE
To hell with get_stacktrace!

### PR DESCRIPTION
**Pull request by NelsonVides**
Now only some dependencies are still giving that silly `get_stacktrace/0 deprecated` warning. To be precise, exactly these dependencies (and dependencies of dependencies and so on):

* worker_pool
* katana
* proper
* pooler
* cqerl
* eper
* nklib
* nkpacket
* fusco
* exometer_core

~Testing on CircleCI is going to be a ton faster than locally~ 😛

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7331.1](https://travis-ci.org/esl/MongooseIM/jobs/598867117) / Erlang 22.0 / small_tests / faa861a49a1692c71f820ea43608851b0d23e9e2
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2494/7331/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2494/7331/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_00.46.04/lib.mongooseim.logs/run.2019-10-17_00.46.04/suite.log.html)

File: src/rdbms/mongoose_rdbms.erl [hunk 1]
```diff
@@ -542,18 +542,18 @@ inner_transaction(F, _State) ->
...
+            {F(), []}
+        catch
+            throw:ThrowResult:S ->
+                {ThrowResult, S};
+            Class:Other:Stacktrace ->
+                ?ERROR_MSG("issue=outer_transaction_failed "
+                           "reason=~p:~p stacktrace=~1000p",
+                           [Class, Other, Stacktrace]),
+                {{'EXIT', Other},Stacktrace}
+        end,
```
Comment (michalwski) [inline src/rdbms/mongoose_rdbms.erl]: A cosmetic thing. I think readability of this block could be improved a bit by moving they `try ... catch` clause to a dedicated function like `apply_transaction_function`. What do you think?

**Comment by mongoose-im** [comment]
[7333.1](https://travis-ci.org/esl/MongooseIM/jobs/599012363) / Erlang 22.0 / small_tests / 8a8a10374d7ffde5b2f1a7c4ed15f8b293e94cb4
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2494/7333/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2494/7333/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_07.32.03/lib.mongooseim.logs/run.2019-10-17_07.32.03/suite.log.html)

**Comment by NelsonVides** [comment]
@michalwski 
> Thanks for the effort! This was quick :) I have only one cosmetic comment to the code. Also there is a small inconsistency in the way variable containing the stack trace is called. Sometimes it's `Stacktrace` and sometimes `StackTrace`. Do you think this naming can be made consistent or is there a reason to have both variants?

I thought the same while doing this, but I pretty much followed the style of the code already present. Sometimes we had pattern matching against `C:R` and sometimes against "Class:Reason`, sometimes the stacktrace was called `StackTrace` and some others it was `Stacktrace`... I didn't know against which to unify. But I'm gonna try `StackTrace` just for the fact that that is actually two glued words and not one alone. I can easily grep over that together with the cosmetic comment :slightly_smiling_face:

**Comment by mongoose-im** [comment]
[7334.1](https://travis-ci.org/esl/MongooseIM/jobs/599044660) / Erlang 22.0 / small_tests / a3ddf3b427869e1c1ed7dd913a814ead243d8987
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2494/7334/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2494/7334/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_08.45.58/lib.mongooseim.logs/run.2019-10-17_08.45.58/suite.log.html)

**Pull request merged**